packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  // The envelope to be sent
+  envelope: Envelope;
+  // A function that returns an event from the envelope if one exists
+  getEvent(): Event | undefined;
+}
+
+type Matcher = (param: MatchParam) => string[];
+
+function eventFromEnvelope(env: Envelope): Event | undefined {
+  let event: Event | undefined;
+
+  forEachEnvelopeItem(env, (item, type) => {
+    if (type === 'event' || type === 'transaction') {
```

Comment:
For now we should guard this so that it is only errors - transactions go to a single DSN.